hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractStoresSubcommand.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.cli.signedstate;
+
+import static com.swirlds.common.threading.manager.AdHocThreadManager.getStaticThreadManager;
+import static java.util.Comparator.naturalOrder;
+import static java.util.Map.Entry.comparingByKey;
+
+import com.hedera.node.app.service.mono.state.virtual.ContractKey;
+import com.hedera.node.app.service.mono.state.virtual.IterableContractValue;
+import com.hedera.services.cli.signedstate.DumpStateCommand.EmitSummary;
+import com.hedera.services.cli.signedstate.DumpStateCommand.WithSlots;
+import com.hedera.services.cli.signedstate.SignedStateCommand.Verbosity;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.BiConsumer;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+
+@SuppressWarnings("java:S106") // "use of system.out/system.err instead of logger" - not needed/desirable for CLI tool
+public class DumpContractStoresSubcommand {
+    static void doit(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path storePath,
+            @NonNull final EmitSummary emitSummary,
+            @NonNull final WithSlots withSlots,
+            @NonNull final Verbosity verbosity) {
+        new DumpContractStoresSubcommand(state, storePath, emitSummary, withSlots, verbosity).doit();
+    }
+
+    @NonNull
+    final SignedStateHolder state;
+
+    @NonNull
+    final Path storePath;
+
+    final EmitSummary emitSummary;
+    final WithSlots withSlots;
+    final Verbosity verbosity;
```

Comment:
Intentionally not private?

Reply:
I don't believe in `private`.  `package private` is totally sufficient for me.  What do you think?

Reply:
Just asking because many parts of the codebase use private variables. This is a deviation from that, and deviations are interesting :)

The only question really worth asking here is if we want to keep the use of `private` relatively consistent throughout the codebase. Will it throw others off if these aren't private? That asked, the answer is 'probably not.' It did catch my eye but doesn't materially change anything about the design. And it certainly won't be enough to hold up this PR 👍